include/caf/events/module_state_event.h

```diff
@@ -19,6 +19,36 @@
 extern "C" {
 #endif
 
+#define MODULE_ID_LIST_SECTION_PREFIX module_id_list
+
+#define MODULE_ID_PTR_VAR(mname) _CONCAT(__module_, MODULE)
+#define MODULE_ID_LIST_SECTION_NAME   STRINGIFY(MODULE_ID_LIST_SECTION_PREFIX)
+#define MODULE_ID_LIST_START _CONCAT(__start_, MODULE_ID_LIST_SECTION_PREFIX)
+#define MODULE_ID_LIST_STOP  _CONCAT(__stop_,  MODULE_ID_LIST_SECTION_PREFIX)
+#define MODULE_ID_PTR_VAR_EXTERN_DEC(mname) \
+	extern const void * const MODULE_ID_PTR_VAR(mname)
+
+extern const void * const MODULE_ID_LIST_START;
+extern const void * const MODULE_ID_LIST_STOP;
+
+static inline int module_count(void)
```

Comment:
please return `size_t`

Reply:
sorry for missing this in the original PR :(

Reply:
Done